src/test.rs

```diff
@@ -447,6 +535,64 @@ impl<W: Worktree + Sync + Send + 'static> Manager<W> {
     }
 }
 
+struct JobDepNotifier {
+    test_case: TestCase,
+    // Used to feed into the overall notification channel for observers to keep
+    // track of what the whole Manager is doing.
+    notif_tx: broadcast::Sender<Arc<Notification>>,
+    // Used to notify specifically about completion of this job. Only one mesage
+    // should be sent on this channel. This is done via a separate channel so
+    // that you can get notified about one job without having to wake up for a
+    // bunch of other unrelated events.
+    completion_tx: watch::Sender<TestStatus>,
+}
+
+impl JobDepNotifier {
+    fn new(test_case: TestCase, notif_tx: broadcast::Sender<Arc<Notification>>) -> Self {
+        // This watch API is a bit weird; it's designed on the assumption that
+        // there is some shared value and you want to be notified when that
+        // value is changed. So when you construct a channel you pass in the
+        // initial value, and the rx will immediately "receive" that value.
+        // That isn't what we're using this for - we just wanna send a single
+        // message, so we pass a random dummy as the "initial" value and drop
+        // the rx. Then to create the actual receivers we call subscribe, since
+        // that doesn't lead to the same weird "channel starts out full"
+        // semantics.
```

Comment:
This was probably dumb, should just have used a broadcast channel. Don't think I care enough to change it now.